python/mxnet/test_utils.py

```diff
@@ -116,7 +116,22 @@ def _validate_csr_generation_inputs(num_rows, num_cols, density,
                              % (density, num_rows, num_cols))
 
 
-def _get_uniform_dataset_csr(num_rows, num_cols, density=0.1, dtype=None):
+def shuffle_csr_column_indices(csr):
+    """Shuffle CSR column indices per row
+    This allows validation of unordered column indices, which is not a requirement
+    for a valid CSR matrix
+    """
+    row_count = len(csr.indptr) - 1
+    for i in range(row_count):
+        start_index = csr.indptr[i]
+        end_index = csr.indptr[i + 1]
+        sublist = np.array(csr.indices[start_index : end_index])
+        np.random.shuffle(sublist)
+        csr.indices[start_index : end_index] = sublist
+
+
+def _get_uniform_dataset_csr(num_rows, num_cols, density=0.1, dtype=None,
```

Comment:
I asked Mu, he said it's okay to support only CSR w/ sorted indices. If an user has unsorted indices, we can provide a sort_indices function for them. This way in backend all operators can assume sorted indices and perform optimization if there's a chance. 
As for this shuffle function, I think it's still useful to unit test the operators that support CSR w/ unsorted indices. But the default value for `shuffle` option should be set to False.